packages/schema/tests/lib/SchemaMiddleware.test.ts

```diff
@@ -0,0 +1,357 @@
+import { MapProvider } from '@joshdb/map';
+import { MathOperator, Method, MiddlewareStore, Payload } from '@joshdb/middleware';
+import { CombinedPropertyError, s } from '@sapphire/shapeshift';
+import { SchemaMiddleware } from '../../src';
+
+describe('SchemaMiddleware', () => {
+  describe('is a class', () => {
+    test('GIVEN typeof SchemaMiddleware THEN returns function', () => {
+      expect(typeof SchemaMiddleware).toBe('function');
+    });
+
+    test('GIVEN typeof ...prototype THEN returns object', () => {
+      expect(typeof SchemaMiddleware.prototype).toBe('object');
+    });
+  });
+
+  describe('can manipulate provider data', () => {
+    // @ts-expect-error 2322
+    const store = new MiddlewareStore({ provider: new MapProvider() });
+    const schema = new SchemaMiddleware<unknown>({
+      schema: s.object({
```

Comment:
is shapeshift exported from the package? otherwise is it something the user has to install manually?

Reply:
They would need to install it separately. I think it's best this way so they can get a better understanding of how it works themselves.

Reply:
Can that be added in the readme, also is it restricted to just shapeshift? because maybe a bettername for the package would be @joshdb/shapeshift ... but not a pressing concern

Reply:
Hmm, that is a good point. I'll add that to the readme.